libnetwork/network.go

```diff
@@ -1524,16 +1524,18 @@ func (n *Network) ipamAllocate() error {
 		}
 	}
 
-	err = n.ipamAllocateVersion(4, ipam)
-	if err != nil {
-		return err
-	}
-
-	defer func() {
+	if n.enableIPv4 {
+		err = n.ipamAllocateVersion(4, ipam)
 		if err != nil {
-			n.ipamReleaseVersion(4, ipam)
+			return err
 		}
-	}()
+
+		defer func() {
+			if err != nil {
```

Comment:
```suggestion
			if retErr != nil {
```

Reply:
So, unrelated to changes in this PR, you're suggesting refactoring this function to make it use a `retErr` - ok, I can do that ...